### DISCUSSION TITLE
Are webpack-obfuscator options syntax exactly the same as javascript-obfuscator?

### DISCUSSION THREAD

**Discussion by 1shevelov** [javascript-obfuscator/javascript-obfuscator#1260] [Q&A]
On the [webpack-obfuscator NPM page](https://www.npmjs.com/package/webpack-obfuscator), there are a couple of examples with this syntax:

> `new WebpackObfuscator ({
>         rotateStringArray: true
>     }, ['excluded_bundle_name.js'])`

and

> `new JavaScriptObfuscator({
>             rotateStringArray: true
>         }, ['abc.js'])`

in **plugins** section of webpack config

**Questions:**
1) Why plugin written as "WebpackObfuscator" in the first example and "JavascriptObfuscator" in the second? What and is there a difference?

2) Option "**rotateStringArray**" used in both examples. But in README of javascript-obfuscator repo there is no such option, but "[stringArrayRotate](https://github.com/javascript-obfuscator/javascript-obfuscator?tab=readme-ov-file#stringarrayrotate)" instead. Is there an error? Or syntax for some options is different?

3) In my project, there is `"webpack-obfuscator": "^3.5.1"` as dev dependency in package.json. But no "javascript-obfuscator" line (I used someone else's boilerplate). Should I install "javascript-obfuscator" as dev dependency also? (obfuscation seems to work alright, but I'm not sure that all options are working)

Please help

**Comment by huuphat1908** [comment]
Do you have any conclusion? In my case, it seems like debugProtection doesn't work
